TextOverflowWrapper/src/TextOverflowWrapper.jsx

```diff
@@ -1,85 +1,88 @@
 import React, { Component, PropTypes } from "react";
 import "./style.less";
 import ReactTooltip from "react-tooltip";
-import ToolTip from "react-portal-tooltip";
 
-const generateID = () => "a" + Math.random().toString(36).substr(2, 10);
-class TextOverflowWrapper extends Component {
 
+class TextOverflowWrapper extends Component {
     constructor() {
         super();
+        this.uniqueId = "overflowTooltip-" + (Date.now() * Math.random());
         this.state = {
-            isTooltipActive: false,
-            id: generateID()
+            itemWidth: 0
         };
     }
-
-    showTooltip() {
-        this.setState({
-            isTooltipActive: true
-        });
+    getStyle() {
+        const {props} = this;
+        return Object.assign({ maxWidth: props.maxWidth }, props.style);
     }
 
-    hideTooltip() {
-        this.setState({
-            isTooltipActive: false
-        });
+    componentDidMount() {
+        //Set time out to ensure calculation happens after render
+        setTimeout(() => {
```

Comment:
Is `setTimeout` the best option to synchronise?

Reply:
Maybe not. I don't know who and why made this. This pull requests focus on project structure and build process.